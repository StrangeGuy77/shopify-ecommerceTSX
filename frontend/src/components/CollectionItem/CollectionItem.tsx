import * as React from 'react';
import './CollectionItem.scss';
import { Items } from '../../pages/ShopPage/ShopPage';
import CustomButton from '../CustomButton/CustomButton';

const CollectionItem: React.FC<Items> = ({ name, price, imageUrl }) => {
    return (
        <div className="collection-item">
            <div className="image" style={{
                backgroundImage: `url(${imageUrl})`
            }} />
            <div className="collection-footer">
                <span className="name">
                    {name}
                </span>
                <span className="price">
                    {price}$
                </span>
            </div>
            <CustomButton inverted> Add to cart </CustomButton>
        </div>
    );
};

export default CollectionItem;