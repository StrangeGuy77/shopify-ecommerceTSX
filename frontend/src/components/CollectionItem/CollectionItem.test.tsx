import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CollectionItem from './CollectionItem';

it('Should render Collection Item correctly', () => {
    const propsTemplate = ({ id, imageUrl, name, price }: Props) => ({
        id, name, imageUrl, price
    });

    const propsToRender = propsTemplate({ id: 1, imageUrl: "", name: "", price: 121 });

    const tree = renderer.create(<CollectionItem {...propsToRender} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

interface Props {
    id: any;
    name: string;
    imageUrl: string;
    price: number;
}