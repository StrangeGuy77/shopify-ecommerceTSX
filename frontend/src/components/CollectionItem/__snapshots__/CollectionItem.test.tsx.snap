// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Collection Item correctly 1`] = `
<div
  className="collection-item"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url()",
      }
    }
  />
  <div
    className="collection-footer"
  >
    <span
      className="name"
    >
      
    </span>
    <span
      className="price"
    >
      121
      $
    </span>
  </div>
  <button
    className="inverted  custom-button"
  >
     Add to cart 
  </button>
</div>
`;
